.travis.yml

```diff
@@ -1,19 +1,40 @@
+sudo: required
+dist: trusty
+group: edge
+addons:
+  apt:
+    packages:
+      - mysql-server-5.6
+      - mysql-client-core-5.6
+      - mysql-client-5.6
+      - postfix
+  firefox: "46.0"
+  hosts:
+    - magento.travis
 language: php
-
 php:
-  - 7.0
-  - 7.1
   - 7.2
   - 7.3
   - 7.4
-
+env:
+  global:
+    - COMPOSER_BIN_DIR=~/bin
+    - MAGENTO_HOST_NAME="magento.travis"
+  matrix:
+    - TEST_SUITE=static
+    - TEST_SUITE=functional
 matrix:
   fast_finish: true
   allow_failures:
-    - php: nightly
-    - php: 7.3
     - php: 7.4
-
+cache:
+  apt: true
+  directories:
+    - $HOME/.composer/cache
+before_install: ./dev/travis/before_install.sh
+install: if [ $TEST_SUITE == "functional" ]; then composer install --no-interaction; fi
+before_script: ./dev/travis/before_script.sh
 script:
-  - '! find . -not \( -path ./.phpstorm.meta.php -prune \) -not \( -path ./lib/PEAR -prune \) -not \( -path ./lib/phpseclib -prune \) -not \( -path ./lib/Zend -prune \) -type f -name "*.php" -exec php -d error_reporting=32767 -l {} \; 2>&1 >&- | grep "^"'
-  - '! find app/design -type f -name "*.phtml" -exec php -d error_reporting=32767 -l {} \; 2>&1 >&- | grep "^"'
+  - if [ $TEST_SUITE == "static" ]; then ! find . -not \( -path ./.phpstorm.meta.php -prune \) -not \( -path ./lib/PEAR -prune \) -not \( -path ./lib/phpseclib -prune \) -not \( -path ./lib/Zend -prune \) -type f -name "*.php" -exec php -d error_reporting=32767 -l {} \; 2>&1 >&- | grep "^"; fi
+  - if [ $TEST_SUITE == "static" ]; then ! find app/design -type f -name "*.phtml" -exec php -d error_reporting=32767 -l {} \; 2>&1 >&- | grep "^"; fi
+  - if [ $TEST_SUITE == "functional" ]; then dev/tests/functional/vendor/phpunit/phpunit/phpunit -c dev/tests/$TEST_SUITE; fi
```

Comment:
Functional tests takes more than 20 mintes and you have to add `travis_wait 30` otherwise task will fail with error: `No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.`

My sugesstion:
```- if [ $TEST_SUITE == "functional" ]; then travis_wait 30 dev/tests/functional/vendor/phpunit/phpunit/phpunit -c dev/tests/$TEST_SUITE; fi```

Reply:
Thanks @szymonbdeligo I'll give it a go!